onnxruntime/core/providers/cuda/tensor/expand.cc

```diff
@@ -72,6 +102,17 @@ Status Expand::ComputeInternal(OpKernelContext* ctx) const {
     return Status::OK();
   }
 
+#ifdef ENABLE_TRAINING
+  // Strided output.
+  if (input_data_tensor.DataRaw() == output_tensor.DataRaw()) {
```

Comment:
I actually have the solution for this in the dev branch. But it will need to change quite some files not related to this PR (for example add func to the bridge thing). Will do this in the PR that moving this to new macro just as what Ashwini suggested.